### PR TITLE
clustectl integration

### PR DESCRIPTION
**Pull request by isaacdorfman**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,7 +36,7 @@ IMAGE_TAG_BASE ?= agent-install.openshift.io/cluster-api-provider-agent
 BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
 
 # Image URL to use all building/pushing image targets
-IMG ?= controller:latest
+IMG ?= quay.io/eranco74/cluster-api-provider-agent@sha256:36285266c563f9421f8b536b99aa69408bf4b828ef5d25479d52089e36d07c31
```
Comment (eranco74) [inline Makefile:39]: Why?
The default should stay the same, just specify the IMG env when invoking the relevant make target.

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -5,6 +5,12 @@ generatorOptions:
   disableNameSuffixHash: true
```
Comment (eranco74) [inline config/manager/kustomization.yaml:5]: No need to commit this file

File: infrastructure-components.yaml [hunk 3]
```diff
@@ -0,0 +1,767 @@
+apiVersion: v1
```
Comment (eranco74) [inline infrastructure-components.yaml:1]: Does this need to be committed?
I think this should get created upon release and saved as part of the release and not as part of the source code

**Comment by eranco74** [comment]
Where is the github action for creating a release?

**Comment by isaacdorfman** [comment]
> Where is the github action for creating a release?

Somehow got delete in rebase.
I restored it now.

File: .github/workflows/create_release.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
```
Comment (eranco74) [inline .github/workflows/create_release.yaml:5]: Please update this to branches that start with `release-`

File: .github/workflows/create_release.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16.6'
+      - name: Create infrastructure components
+        run: |
+          make create_infrastructure-components
```
Comment (eranco74) [inline .github/workflows/create_release.yaml:22]: You should specify the `IMG` env

File: Makefile [hunk 6]
```diff
@@ -37,6 +37,7 @@ BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
 
 # Image URL to use all building/pushing image targets
 IMG ?= controller:latest
+CLUSTERCTL_IMG ?= quay.io/eranco74/cluster-api-provider-agent@sha256:36285266c563f9421f8b536b99aa69408bf4b828ef5d25479d52089e36d07c31
```
Comment (eranco74) [inline Makefile:40]: This is still hardcoded, why not set the relevant image in the `IMG` env when calling `create_infrastructure-components`?

File: .github/workflows/create_release.yaml (hunk 5 above)
Comment (eranco74) [reply .github/workflows/create_release.yaml:22]: Something like this:
```suggestion
          make create_infrastructure-components IMG="quay.io/eranco74/cluster-api-provider-agent:${{ steps.vars.outputs.sha_short }}
```

File: Makefile (hunk 6 above)
Comment (eranco74) [reply Makefile:40]: @isaacdorfman ?

**Review by eranco74 — APPROVED** [review]

**Pull request merged**